client.lua

```diff
@@ -158,6 +159,16 @@ local function StartFueling(vehicle, fuelingMode)
         price = 0
         moneyAmount = exports.ox_inventory:Search(2, 'money')
     end
+
+    if fuelingMode == 'can' then 
+        local fuelcan = exports.ox_inventory:Search('slots', 'WEAPON_PETROLCAN')
```

Comment:
so event handler is just to get the slot of the item in hand. then i just compare the slot number to get the right can, if multiple are in my inventory, so i can pull durability metadata out then thats used to compare against the fuelamount so it can cancel fuelling the can. otherwise the player will still do the whole progress bar. i thought about passing metadata from the currentweapon event handler but then the data could be stale since it is only updated once a weapon is holstered or pulled out.